cremalab/crema-components
# Tabs component

## Description

Adds a `<Tabs />` component for displaying content.

### Usage

```tsx
<Tabs currentTab="1" onTab={(nextTab) => setCurrentTab(nextTab)}>
  <Tab id="1" label="Tab 1">Content 1</Tab>
  <Tab id="2" label="Tab 2" disabled>Content 2</Tab>
  <Tab id="3" label="Tab 3">Content 3</Tab>
</Tabs>
```

If a user passes a bad initial `id` (one that doesn't exist OR belongs to a disabled tab), we simply throw an error telling them what's wrong.

### 📋 Acceptance Criteria

> _Checked off by PR **Assignees**_

- [x] Tabs can be composed using both `Tabs` and `Tab` components
- [x] When a "tab" is clicked, its content is displayed
- [x] An "initial tab" index can be set
- [x] An `onTab` function is called with the current index when a tab is selected
- [x] A11y tags have been appropriately applied
- [x] Keyboard events and tabbing is supported
- [x] Supports "disabled" tabs
- [x] Throws errors for bad ID or disabled tab

### 🖼 Demo

https://user-images.githubusercontent.com/157195/180878731-72f1cd64-8c70-4246-8b8c-db80525a6fd5.mov

#### With Disabled Tabs

https://user-images.githubusercontent.com/157195/182234524-a5054b3e-e716-4798-8195-cadff6a9bbbd.mov

### 👩‍🔬 Test Instructions

> _Provided by PR **Assignees**_

1. Install dependencies
2. `npm run test:playground`
3. View "Tabs" stories
4. Select various tabs and ensure that corresponding content is displayed5. 
5. Utilize Tab and Arrow Left/Right keys to navigate

## 🔎 Reviewer Checklist

> _Checked off by PR **Reviewers**_




---

> ### Roles & Responsibilities
>
> #### 👨‍💻 Assignee
>
> - Initiator of this PR (be sure to set in GitHub UI)
> - Addresses feedback and change requests
> - Merges PR once approved (usually deletes branch unless `develop` or `release`)
>
> #### 👩‍💻 Reviewer
>
> - Invited to review PR by **Assignee** (via GitHub UI)
> - Is expected to complete a review and address followup
